lenskit-knn/src/main/java/org/lenskit/knn/item/ItemItemItemBasedItemScorer.java

```diff
@@ -21,28 +21,34 @@
 package org.lenskit.knn.item;
 
 import it.unimi.dsi.fastutil.longs.Long2DoubleMap;
-import it.unimi.dsi.fastutil.longs.LongSortedSet;
-import org.grouplens.lenskit.basic.AbstractGlobalItemScorer;
+import it.unimi.dsi.fastutil.longs.LongIterator;
+import it.unimi.dsi.fastutil.longs.LongIterators;
+import it.unimi.dsi.fastutil.longs.LongSet;
 import org.grouplens.lenskit.util.ScoredItemAccumulator;
 import org.grouplens.lenskit.util.TopNScoredItemAccumulator;
 import org.grouplens.lenskit.util.UnlimitedScoredItemAccumulator;
-import org.grouplens.lenskit.vectors.MutableSparseVector;
 import org.grouplens.lenskit.vectors.SparseVector;
 import org.grouplens.lenskit.vectors.VectorEntry;
+import org.lenskit.api.Result;
+import org.lenskit.api.ResultMap;
+import org.lenskit.basic.AbstractItemBasedItemScorer;
 import org.lenskit.knn.NeighborhoodSize;
 import org.lenskit.knn.item.model.ItemItemModel;
+import org.lenskit.results.Results;
 import org.lenskit.util.collections.LongUtils;
 
 import javax.annotation.Nonnull;
 import javax.inject.Inject;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.List;
 
 /**
  * Score items based on the basket of items using an item-item CF model.
  *
  * @author <a href="http://www.grouplens.org">GroupLens Research</a>
  */
-public class ItemItemGlobalScorer extends AbstractGlobalItemScorer {
+public class ItemItemItemBasedItemScorer extends AbstractItemBasedItemScorer {
```

Comment:
Yes, it does.

It sounds like something Doctor Donna might have said as her brain was snapping from time-lord-ish-ness.

We have an outstanding inconsistency, too: our item-item CF `ItemScorer` is called `ItemItemScorer` rather than `ItemItemItemScorer`.

Way to work around this problem: introduce a notion of item similarity (other than the one used for computing the matrices?) and pull this out to be a `SimilarityItemBasedItemScorer` (since that's really what it is), and have item-item CF provide an item-item similarity that pulls from the model.